tests/testsuite/support/matchers.rs

```diff
@@ -0,0 +1,734 @@
+use std::fmt;
+use std::process::Output;
+use std::str;
+use std::usize;
+
+use serde_json::{self, Value};
+
+use support::hamcrest;
+use support::process::{ProcessBuilder, ProcessError};
+
+#[derive(Clone)]
+pub struct Execs {
+    expect_stdout: Option<String>,
+    expect_stdin: Option<String>,
+    expect_stderr: Option<String>,
+    expect_exit_code: Option<i32>,
+    expect_stdout_contains: Vec<String>,
+    expect_stderr_contains: Vec<String>,
+    expect_either_contains: Vec<String>,
+    expect_stdout_contains_n: Vec<(String, usize)>,
+    expect_stdout_not_contains: Vec<String>,
+    expect_stderr_not_contains: Vec<String>,
+    expect_stderr_unordered: Vec<String>,
+    expect_neither_contains: Vec<String>,
+    expect_json: Option<Vec<Value>>,
+}
+
+impl Execs {
+    /// Verify that stdout is equal to the given lines.
+    /// See `lines_match` for supported patterns.
+    #[allow(dead_code)]
+    pub fn with_stdout<S: ToString>(mut self, expected: S) -> Execs {
+        self.expect_stdout = Some(expected.to_string());
+        self
+    }
+
+    /// Verify that stderr is equal to the given lines.
+    /// See `lines_match` for supported patterns.
+    #[allow(dead_code)]
+    pub fn with_stderr<S: ToString>(mut self, expected: S) -> Execs {
+        self._with_stderr(&expected);
+        self
+    }
+
+    #[allow(dead_code)]
+    fn _with_stderr(&mut self, expected: &ToString) {
+        self.expect_stderr = Some(expected.to_string());
+    }
+
+    /// Verify the exit code from the process.
+    #[allow(dead_code)]
```

Comment:
This isn't dead either, is it? It's worth doing a pass through and removing any unnecessary `#[allow(...)]` attributes.

Reply:
extracting to a crate allowed removing all the `dead_code` attributes